crates/core/src/db/relational_db.rs

```diff
@@ -95,33 +95,56 @@ impl RelationalDB {
             let mut last_commit_offset = None;
             let mut last_hash: Option<Hash> = None;
             if let Some(message_log) = &message_log {
-                let message_log = message_log.lock().unwrap();
+                let mut message_log = message_log.lock().unwrap();
                 let max_offset = message_log.open_segment_max_offset;
-                for commit in commit_log::Iter::from(message_log.segments()) {
-                    let commit = commit?;
-
-                    segment_index += 1;
-                    last_hash = commit.parent_commit_hash;
-                    last_commit_offset = Some(commit.commit_offset);
-                    for transaction in commit.transactions {
-                        transaction_offset += 1;
-                        // NOTE: Although I am creating a blobstore transaction in a
-                        // one to one fashion for each message log transaction, this
-                        // is just to reduce memory usage while inserting. We don't
-                        // really care about inserting these transactionally as long
-                        // as all of the writes get inserted.
-                        datastore.replay_transaction(&transaction, odb.clone())?;
-
-                        let percentage = f64::floor((segment_index as f64 / max_offset as f64) * 100.0) as i32;
-                        if percentage > last_logged_percentage && percentage % 10 == 0 {
-                            last_logged_percentage = percentage;
-                            log::debug!(
-                                "[{}] Loaded {}% ({}/{})",
-                                address,
-                                percentage,
-                                transaction_offset,
-                                max_offset
-                            );
+                let total_segments = message_log.total_segments();
+
+                'replay: for (segment_offset, segment) in message_log.segments().enumerate() {
```

Comment:
Note that I intend to reduce the rightward drift here and move the replay logic elsewhere.
I left it here for this patch to preserve the side-by-side diff for review.